include/dictionary-search.php

```diff
@@ -118,7 +115,7 @@ function is_match_whole_words($search)
 	if(!isset($_GET['partialsearch']))
 	{
 		$partialsearch = get_option("include_partial_words");
-		if($partialsearch == 1 && $wp_query->query_vars['match_whole_words'] == 0)
+		if($partialsearch == 1 && $match_whole_words == 0)
```

Comment:
This `if` block does nothing.

Reply:
Good catch. I think the logic was wrong in the first place. Should have been:

```
		$partialsearch = get_option("include_partial_words");
		if($partialsearch == 1 && $match_whole_words == 1)
		{
			$match_whole_words = 0;
		}
```

So include_partial_words would override match_whole_words

Reply:
Ah, I think the logic was wrong. include_partial_words option says "Always include partial words in search", so it would override the previous match_whole_words setting.

Reply:
Changed to:
```

		if($partialsearch == 1 && $match_whole_words == 1)
		{
			$match_whole_words = 0;
		}
```